core/src/repair/serve_repair_service.rs

```diff
@@ -53,13 +59,37 @@ impl ServeRepairService {
             socket_addr_space,
             Some(stats_reporter_sender),
         );
-        let t_listen = serve_repair.listen(blockstore, request_receiver, response_sender, exit);
+        let t_listen =
+            serve_repair.listen(blockstore, remote_request_receiver, response_sender, exit);
 
-        let thread_hdls = vec![t_receiver, t_responder, t_listen];
+        let thread_hdls = vec![t_receiver, t_packet_adapter, t_responder, t_listen];
         Self { thread_hdls }
     }
 
     pub(crate) fn join(self) -> thread::Result<()> {
         self.thread_hdls.into_iter().try_for_each(JoinHandle::join)
     }
 }
+
+// Adapts incoming UDP repair requests into RemoteRequest struct.
+pub(crate) fn adapt_repair_requests_packets(
+    packets_receiver: Receiver<PacketBatch>,
+    remote_request_sender: Sender<RemoteRequest>,
+) {
+    for packets in packets_receiver {
+        for packet in &packets {
+            let Some(bytes) = packet.data(..).map(Vec::from) else {
```

Comment:
Can we directly take the packet and send to RemoteRequest to reduce data copy?

Reply:
The rest of the processing pipeline is updated to process on `RemoteRequest`, so this type change here is necessary for as long as we are going to support the UDP path. This code will be removed once UDP path is turned off.
In the meanwhile, the requests are relatively small, and need to be sigverified and deserialized. So the data copy here really wouldn't move a needle.